engine/src/main/java/org/terasology/world/block/family/MultiConnectFamily.java

```diff
@@ -36,160 +34,62 @@
 import org.terasology.world.block.loader.BlockFamilyDefinition;
 import org.terasology.world.block.shapes.BlockShape;
 
-import java.util.Collection;
 import java.util.Locale;
-import java.util.Map;
 import java.util.Set;
-import java.util.stream.Collectors;
 
-@BlockSections({"no_connections", "one_connection", "line_connection", "2d_corner", "3d_corner", "2d_t", "cross", "3d_side", "five_connections", "all"})
 public abstract class MultiConnectFamily extends AbstractBlockFamily implements UpdatesWithNeighboursFamily {
```

Comment:
Missing JavaDocs :)

This class is pretty esoteric so a bit of documentation about why it's abstract and the contracts for its abstract methods could go a long way. I think generalizing it is the right way to go, however.

Otherwise I think it needs a generously commented concrete implementation.